source/agora/common/Amount.d

```diff
@@ -259,6 +259,11 @@ pure @safe nothrow @nogc unittest
 
     assert(Amount(100_500_000).integral() == 10);
     assert(Amount(100_500_000).decimal() == 500_000);
+
+    assert(Amount.FreezeAmount.decimal() == 0);
+    assert(Amount(400_000_000_000, true).integral() == Amount.FreezeAmount.integral());
```

Comment:
The decimal test is good, but testing a constant is rarely adding value, as those rarely change.
I would just replace this and following lines by `assert(Amount.FreezeAmount.integral() == 40_000);`, which is testing an expectation with the integral function.

Reply:
applied the suggest.